info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -261,18 +270,46 @@ void storeElectionContestInDatabase(String electionQueryId, JsonObject contest)
     datastore.update(electionEntity);
   }
 
-  // @TODO [Find incumbency.]
+  private boolean checkIncumbency(String candidateName, String position, String division) {
+    try {
+      String queryUrl = String.format("%s&ocdDivisionId=%s", REPRENTATIVE_QUERY_URL, division);
+      JsonObject representatives = queryCivicInformation(queryUrl);
+      JsonArray offices = representatives.getAsJsonArray("offices");
+      int i = 0;
+      for (JsonElement eachOffice : offices){
+        JsonObject office = (JsonObject) eachOffice;
+        String officeName = office.get("name").getAsString();
+        if (position.equals(officeName)){
+          break;
+        }
+        i++;
+      }
+      JsonObject officials = representatives.getAsJsonObject("officials");
+      String incumbent = (officials.get("name").getAsString());
+      return candidateName.equals(incumbent);
+    } catch (IOException e) {
+        System.out.println(
+          String.format(
+              "[ERROR] Failed to query the Civic Information API for %s: %s.", "representatives", e));
+      return false;
+    }
+  }
+
   /**
    * Stores information of a {@code candidate} in the database, and updates the candidate's running
    * position's information for the election. Information includes: name, party affiliation,
    * incumbency status, and news articles related to the candidate.
    */
   private void storeElectionContestCandidateInDatabase(JsonObject candidate,
-      List<Value<String>> candidateIds, List<Value<Boolean>> candidateIncumbency) {
+      List<Value<String>> candidateIds, List<Value<Boolean>> candidateIncumbency, 
+      List<Value<String>> candidatePositions, Value<String> position, String ocdDivisionId) {
     String name = candidate.get("name").getAsString();
     String party = candidate.get("party").getAsString();
-    // @TODO [May expand to other information to uniquely identify a candidate. Currently,
-    // candidate information includes only name and party affiliation.]
+    String email = getField(candidate, "email");
+    String phone = getField(candidate, "phone");
+    String photo = getField(candidate, "PhotoUrl");
```

Comment:
nit: "photoUrl". But feel free to leave it as it is if you don't want to change it.